packages/commons/src/auth/token/interopTokenGenerator.ts

```diff
@@ -0,0 +1,126 @@
+/* eslint-disable max-params */
+import { Algorithm, JwtHeader, JwtPayload } from "jsonwebtoken";
+import { v4 as uuidv4 } from "uuid";
+import { userRoles } from "../authData.js";
+import { buildPrivateKeysKidHolder } from "../keys/keyHolder.js";
+import { buildSignerService } from "../signerService.js";
+import { logger } from "../../index.js";
+import { jwtSeedConfig } from "../../config/jwtConfig.js";
+import { InternalToken, TokenSeed } from "./token.js";
+
+export type InteropTokenGenerator = {
+  generateInternalToken: (
+    subject: string,
+    audience: string[],
+    tokenIssuer: string,
+    secondsDuration: number
+  ) => Promise<InternalToken>;
+};
+
+const createInternalTokenWithKid = (
+  algorithm: Algorithm,
+  kid: string,
+  subject: string,
+  audience: string[],
+  tokenIssuer: string,
+  validityDurationSeconds: number
+): TokenSeed => {
+  const issuedAt = new Date();
+  return {
+    id: uuidv4(),
+    algorithm,
+    kid,
+    subject,
+    issuer: tokenIssuer,
+    issuedAt: issuedAt.getTime() / 1000,
+    nbf: issuedAt.getTime() / 1000,
+    expireAt: issuedAt.setSeconds(
+      issuedAt.getSeconds() + validityDurationSeconds
+    ),
+    audience,
+    customClaims: new Map([["role", userRoles.INTERNAL_ROLE]]),
+  };
+};
+
+export const buildInteropTokenGenerator = (): InteropTokenGenerator => {
+  // Hosting all the dependencies to collect all process env reading at one time
+  const kidHolder = buildPrivateKeysKidHolder();
+  const signerService = buildSignerService();
+  const { subject, audience, tokenIssuer, secondsToExpire } = jwtSeedConfig();
```

Comment:
In this scenario I prefer to hosting all configurations in a closure scope to avoid more reads ops every time when `createSignedJWT` is called.